mne_icalabel/features/topomap.py

```diff
@@ -0,0 +1,121 @@
+import numpy as np
+from mne.channels.layout import _find_topomap_coords
+from mne.defaults import _BORDER_DEFAULT, _EXTRAPOLATE_DEFAULT, _INTERPOLATION_DEFAULT
+from mne.io import Info
+from mne.io.pick import _get_channel_types, _pick_data_channels, pick_info
+from mne.preprocessing import ICA
+from mne.viz.topomap import _check_extrapolate, _make_head_outlines, _setup_interp
+from mne.viz.utils import _setup_vmin_vmax
+
+
+def get_topo_array(ica: ICA, picks="eeg"):
+    """Generate an array of scalp topographies plot of size (n_pixels, n_pixels).
+
+    Parameters
+    ----------
+    ica : ICA
+        Instance of MNE ICA Decomposition mne.preprocesssing.ICA.
+    picks : str | list | slice | None
+        In lists, channel type strings (e.g., ['meg', 'eeg']) will pick channels
+        of those types, channel name strings (e.g., ['MEG0111', 'MEG2623'] will
+        pick the given channels. Can also be the string values “all” to pick all
+        channels, or “data” to pick data channels. None (default) will pick all
+        channels. Note that channels in info['bads'] will be included if their
+        names or indices are explicitly provided.
+
+    Returns
+    -------
+    topo_array : np.ndarray of shape (n_components, n_pixels, n_pixels)
+    """
```

Comment:
```suggestion
def get_topomaps(ica: ICA, picks=None):
    """Generate an array of scalp topographies (n_pixels, n_pixels) for the picked components.

    Parameters
    ----------
    ica : ICA
        Instance of MNE `~mne.preprocesssing.ICA` decomposition.
    %(picks_ica)s ``None`` (default) will pick all sources in the order fitted.

    Returns
    -------
    topomaps : array of shape (n_components, n_pixels, n_pixels)
    """
```

Let's try to be consistent with the names to make the API simpler:
`topomap.py` so let's consider this feature's name as 'topomap'. Let's go for:

- `get_topomap` for single IC -> (n_pixels, n_pixels)
- `get_topomaps` for multiple ICs at once -> (n_components, n_pixels, n_pixels)

Also, the `picks` here should refer to the `picks_ica` to pick ICs: https://github.com/mne-tools/mne-python/blob/e8ded9719de7425c8d6752d2942a6ecba20fd0fc/mne/utils/docs.py#L2289-L2292
So the default to `eeg` and the docstring is wrong as it is not used to pick data channels but ICs. I think it's also missing the logic below to use the picks. Those lines should be adapted:

```
    data = np.dot(
        ica.mixing_matrix_[:, : ica.n_components_].T,
        ica.pca_components_[: ica.n_components_],
    )
```

One more note, in the docstring I changed the ICA parameter line to 
```
Instance of MNE `~mne.preprocesssing.ICA` decomposition.
``` 

which will render as `Instance of MNE ICA decomposition.` with a x-link to the ICA API documentation. The x-link is handled by the backticks, and the tilde `~` removes the parent modules (mne, preprocessing) and only keeps the class/function/method name.

Reply:
I also renamed the returned variable to `topomaps`, please update the variable name in the code accordingly.